lib/omnibus/fetchers/net_fetcher.rb

```diff
@@ -125,6 +125,11 @@ def checksum
       source[digest_type]
     end
 
+    # Set to false to extract archives into the root source_dir (default).
+    # Set to true to always extract underneath project_dir
+    #
+    attr_accessor :isolate_extraction
```

Comment:
I'm convinced.

The extra piece of info that swayed me was realizing that the comments below for line 218 where the code was mistakenly checking if project_dir was a directory instead of downloaded_file (project_dir is always going to be a directory) don't even matter.  We were always expecting to copy stuff into a directory and other software definitions like cacert.rb were written to work around this (which I found out while attempting a test build).  So this was always a bit wonky and not doing the right thing anyway.

I'll go clean up this PR with a chainsaw now.